lib/_http_client.js

```diff
@@ -182,7 +182,7 @@ function ClientRequest(options, cb) {
                       'client');
     }
     self._storeHeader(self.method + ' ' + self.path + ' HTTP/1.1\r\n',
-                      self._headers);
+                      self._headersStore);
```

Comment:
@targos @mscdex this matches my findings. In practice I did not see any perf impact in using symbols since v7.